cache.go

```diff
@@ -117,32 +125,37 @@ func (cache *Cache) DelInt(key int64) (affected bool) {
 }
 
 func (cache *Cache) EvacuateCount() (count int64) {
-	for i := 0; i < 256; i++ {
+	for i := 0; i < SEGMENT_NUMBER; i++ {
 		count += atomic.LoadInt64(&cache.segments[i].totalEvacuate)
 	}
 	return
 }
 
 func (cache *Cache) ExpiredCount() (count int64) {
-	for i := 0; i < 256; i++ {
+	for i := 0; i < SEGMENT_NUMBER; i++ {
 		count += atomic.LoadInt64(&cache.segments[i].totalExpired)
 	}
 	return
 }
 
 func (cache *Cache) EntryCount() (entryCount int64) {
-	for i := 0; i < 256; i++ {
-		entryCount += atomic.LoadInt64(&cache.segments[i].entryCount)
+	it := cache.NewIterator()
+	for {
+		entry := it.Next()
+		if entry == nil {
+			break
+		}
+		entryCount = atomic.AddInt64(&entryCount, 1)
```

Comment:
Why atomic.AddUint64 ? There is no concurrent access to local variable.

Why this function were changed at all? Can you prove there is a need to change this function?